### ISSUE TITLE
proof request attribute groupName is the same as another one

### ISSUE THREAD

**Issue by blu3beri** [hyperledger/aries-framework-javascript#586]
When we receive a proofrequest it should not be possible for any `groupName` to be the same. e.g. When we receive a proof request with groupName `foo`, `bar` and `foo` it will not retrieve the correct credentials from Indy as we include `foo` twice.

This should be fixed IMO when generating a proof request on the verifier side. However it might be very useful to check if there are non-unique keys when receiving a proof request and logging a warning or handling it at least in some way. 

tl;dr 

We can receive proof request with non-unique `groupNames`. This will result in unexpected behaviour and we will not be able to fulfil  the proof request.

Places we need to check for the proof request keys: (proofService)
- When we receive a proof request
  - processRequest in service
- When we send a proof request
   - createRequest
   - createRequestAsResponse 

What to check:
- Check whether proof request attributes keys and predicate keys do not overlap.
